#1604524370 Design page to view and edit profile

blackshady opened the pull request:
#### What does this PR do?
  Implement Page for the user to be able to view and edit their profile

#### Description of Task to be completed?

- [x] Build out UI for the user to view and edit their profile
- [x] Build out functionality for the user to opt in and out of notification
- [x] Wrote test case

 

#### How should this be manually tested?
- git fetch 
- Checkout to this branch `git checkout ft-1604524370-view-edit-profile-page`
- add the following to env file. `REACT_APP_CLOUDINARY_URL` `REACT_APP_CLOUDINARY_UPLOAD_PRESET`
            `REACT_APP_CLOUDINARY_API_KEY`
- RUN `npm start`
- Sigin up or Login
- Navigate to [http://localhost:3000/dashboard](http://localhost:3000/dashboard) 

#### Any background context you want to provide?
Make sure to set env variable `REACT_APP_BASE_API_URL` to the backend server address
#### What are the relevant pivotal tracker stories?
- [#1604524370](https://www.pivotaltracker.com/story/show/160452437)
- [#160452447](https://www.pivotaltracker.com/story/show/160452447)
#### Screenshots (if appropriate)

<img width="1430" alt="screen shot 2018-11-15 at 11 58 33 am" src="https://user-images.githubusercontent.com/26234445/48549617-8f4f9880-e8d0-11e8-976e-62672c171331.png">

#### Questions:

waleCloud commented:
Hello @blackshady Well done on raising a PR. However, I noticed that your PR message didn't quite has some of the context heading information required such as;

- How should this be manually tested? - which I believe should be available.
- Any background context you want to provide?
- Screenshots (if appropriate) - which I believe is available as the task being a frontend implementation.

waleCloud commented on src/hoc/asyncComponent/asyncComponent.js:
```diff
@@ -0,0 +1,22 @@
+import React, { Component } from 'react';
+
+const asyncComponent = importComponent => class extends Component {
+        state = {
```
@blackshady I want to believe state management should be done with redux alone.

waleCloud reviewed (commented):
Endeavor to add relevant tests & increase test coverage, endeavor to cover all necessary files created/modified. before raising a final PR.
well done so far! 👍

blackshady commented:
Thank you for the feedback @waleCloud. I will add the relevant test and finish up on the feaure ASAP.

uchemukolo commented on src/components/dashboard/DashboardContainer.jsx:
```diff
@@ -0,0 +1,58 @@
+import React, { PureComponent } from 'react';
```
@blackshady I noticed you used `PureComponent` instead of `Component`, is there a particular reason?

blackshady replied:
PureComponent is exactly the same as Component except that it handles the shouldComponentUpdate method for you.

uchemukolo reviewed (commented):
Great job @blackshady for implementing this feature. I have tested the features and everything is working fine. However, the test coverage is less than the 80% threshold set. Try to improve it.

blackshady committed:
feat(Profile Page): View and edit profile page
- Create page to view and edith user profile details
- build out functionality for user to opt in and out of notification
- Create functionality for user to uplod image
- Create test case
[Delivers #1604524370]
